sdk/communication/AzureCommunicationChat/Source/Signaling/CommunicationSignalingClient.swift

```diff
@@ -98,18 +135,73 @@ class CommunicationSignalingClient {
     }
 
     func on(event: ChatEventId, handler: @escaping TrouterEventHandler) {
+        // Guard to check if selfHostedTrouterClient is initialized
+        guard let selfHostedTrouterClient = self.selfHostedTrouterClient else {
+            logger.error("Failed to register event handler: SelfHostedTrouterClient is not initialized.")
+            return
+        }
+        
         let logger = ClientLoggers.default(tag: "AzureCommunicationHandler-\(event)")
         let communicationHandler = CommunicationHandler(handler: handler, logger: logger)
         selfHostedTrouterClient.register(communicationHandler, forPath: "/\(event)")
         communicationHandlers[event] = communicationHandler
     }
 
     func off(event: ChatEventId) {
+        // Guard to check if selfHostedTrouterClient is initialized
+        guard let selfHostedTrouterClient = self.selfHostedTrouterClient else {
+            logger.error("Failed to unregister event handler: SelfHostedTrouterClient is not initialized.")
+            return
+        }
+        
         if let communicationHandler = communicationHandlers[event] {
             selfHostedTrouterClient.unregisterListener(communicationHandler)
             communicationHandlers.removeValue(forKey: event)
         }
     }
+    
+    private func getTrouterSettings(
+            token: String,
+            endpoint: String,
+            completionHandler: @escaping (Result<RealTimeNotificationConfiguration, AzureError>) -> Void
+        ) {
+            let urlString = "\(endpoint)/chat/config/realTimeNotifications?api-version=2024-09-01"
+            guard let url = URL(string: urlString) else {
+                completionHandler(.failure(AzureError.client("Invalid URL constructed for real-time notifications settings.")))
+                return
+            }
+
+            var request = URLRequest(url: url)
+            request.httpMethod = "GET"
+            request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+
+            let session = URLSession.shared
+            session.dataTask(with: request) { data, response, error in
+                if let error = error {
```

Comment:
Please throw communication error

Reply:
Yes we throw AzureError here, which is the type of error thrown in other places within the project.